src/components/data-exploration/heatmap/HeatmapPlot.jsx

```diff
@@ -231,29 +282,79 @@ const HeatmapPlot = (props) => {
     );
   }
 
-  if (!vegaData) {
+  if (!heatmapData) {
     return (
       <center style={{ marginTop: height / 2 }}>
         <Skeleton.Image />
       </center>
     );
   }
 
-  const signalListeners = {
-    mouseOver: handleMouseOver,
+  const setTrackHighlight = (info) => {
+    if (!info) {
+      setHighlightedTrackData(null);
+      return;
+    }
+
+    dispatch(updateCellInfo({ cellName: info[0] }));
+
+    const [cellIndexStr, trackIndex, mouseX, mouseY] = info;
+
+    const trackOrder = Array.from(heatmapSettings.selectedTracks).reverse();
+
+    const cellSetClassKey = trackOrder[trackIndex];
+    const cellClassProps = getCellClassProperties(
+      parseInt(cellIndexStr, 10), cellSetClassKey,
+      cellSetsHierarchy, cellSetsProperties,
+    );
+
+    const obj = {
+      cellId: cellIndexStr,
+      trackName: cellClassProps?.name,
+      coordinates: { x: mouseX, y: mouseY },
+    };
+
+    setHighlightedTrackData(obj);
   };
 
   return (
-    <div>
-      <VegaHeatmap
-        spec={vegaSpec}
-        data={vegaData}
-        showAxes={selectedGenes?.length <= 30}
-        rowsNumber={selectedGenes.length}
-        signalListeners={signalListeners}
-        width={width}
+    <div id='heatmap-container'>
+      <Heatmap
+        uuid='heatmap-0'
+        theme='light'
+        width={width - heatmapRightMargin}
         height={height}
+        colormap='plasma'
+        colormapRange={[0.0, 1.0]}
+        expressionMatrix={vitessceData?.expressionMatrix}
+        cellColors={vitessceData?.metadataTracks.dataPoints}
+        cellColorLabels={vitessceData?.metadataTracks.labels}
+        transpose
+        viewState={viewState}
+        setViewState={({ zoom, target }) => { setViewState({ zoom, target }); }}
+        setCellHighlight={setCellHighlight}
+        setGeneHighlight={setGeneHighlight}
+        setTrackHighlight={setTrackHighlight}
+        updateViewInfo={updateCellCoordinates}
       />
+      <div>
+        {
+          highlightedTrackData ? (
+            <HeatmapTracksCellInfo
+              cellId={highlightedTrackData.cellId}
+              trackName={highlightedTrackData.trackName}
+              coordinates={highlightedTrackData.coordinates}
+            />
+          ) : (
+            <HeatmapCellInfo
+              cellId={cellHighlight}
+              geneName={geneHighlight}
+              geneExpression={focusedExpression?.rawExpression.expression[cellHighlight]}
+              coordinates={cellCoordinatesRef.current}
+            />
+          )
+        }
+      </div>
     </div>
```

Comment:
isn't there a situation when you don't want to display any of the two cell info components, but just nothing?

Reply:
This case is handled inside `HeatmapCellInfo`, but you're right in that that case's handling it should be more explicit

Reply:
it will make more sense if it is handled from here